stl/inc/ranges

```diff
@@ -216,12 +229,21 @@ namespace ranges {
         range_difference_t<_Rng> _Off = -1;
 
     protected:
-        _Cached_position()  = default;
-        ~_Cached_position() = default;
+        constexpr _Cached_position()  = default;

```

Comment:
I am oh so slightly unsure about this. It does enforce constexpr but we could remove it. 

It does create a bit of divergence between the different specializations of we drop it here

Reply:
I did go ahead and strip these `constexpr`s. I won't complain if you put them back =)